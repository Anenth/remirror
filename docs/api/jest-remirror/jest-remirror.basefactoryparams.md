<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](./index.md) &gt; [jest-remirror](./jest-remirror.md) &gt; [BaseFactoryParams](./jest-remirror.basefactoryparams.md)

## BaseFactoryParams interface

<b>Signature:</b>

```typescript
export interface BaseFactoryParams<GSchema extends EditorSchema = EditorSchema> extends Partial<AttrsParams> 
```

## Properties

|  Property | Type | Description |
|  --- | --- | --- |
|  [name](./jest-remirror.basefactoryparams.name.md) | <code>string</code> | The name of the node or mark |
|  [schema](./jest-remirror.basefactoryparams.schema.md) | <code>GSchema</code> | The editor schema |

