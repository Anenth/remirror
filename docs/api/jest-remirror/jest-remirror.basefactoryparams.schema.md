<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](./index.md) &gt; [jest-remirror](./jest-remirror.md) &gt; [BaseFactoryParams](./jest-remirror.basefactoryparams.md) &gt; [schema](./jest-remirror.basefactoryparams.schema.md)

## BaseFactoryParams.schema property

The editor schema

<b>Signature:</b>

```typescript
schema: GSchema;
```
