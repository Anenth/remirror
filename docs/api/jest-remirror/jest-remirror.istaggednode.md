<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](./index.md) &gt; [jest-remirror](./jest-remirror.md) &gt; [isTaggedNode](./jest-remirror.istaggednode.md)

## isTaggedNode variable

Check if a node is tagged.

<b>Signature:</b>

```typescript
isTaggedNode: (val: unknown) => val is TaggedProsemirrorNode<import("@remirror/core").EditorSchema<string, string>>
```
