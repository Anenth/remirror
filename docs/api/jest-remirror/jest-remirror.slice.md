<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](./index.md) &gt; [jest-remirror](./jest-remirror.md) &gt; [slice](./jest-remirror.slice.md)

## slice variable

<b>Signature:</b>

```typescript
slice: (schema: EditorSchema<string, string>) => (...content: TaggedContentWithText[]) => Slice<EditorSchema<string, string>>
```
