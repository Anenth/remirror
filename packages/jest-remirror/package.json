{
  "name": "jest-remirror",
  "version": "0.13.1",
  "description": "A text editor for react built with prosemirror",
  "keywords": [
    "prosemirror",
    "react",
    "ssr",
    "text editor"
  ],
  "homepage": "https://github.com/remirror/remirror/tree/next/packages/jest-remirror",
  "repository": "https://github.com/remirror/remirror/tree/master/packages/jest-remirror",
  "license": "MIT",
  "contributors": [
    "Ifiok Jr. <ifiokotung@gmail.com>"
  ],
  "main": "dist/jest-remirror.cjs.js",
  "module": "dist/jest-remirror.esm.js",
  "types": "dist/jest-remirror.cjs.d.ts",
  "files": [
    "dist",
    "environment"
  ],
  "dependencies": {
    "@babel/runtime": "^7.10.1",
    "@remirror/core": "^0.11.0",
    "@remirror/pm": "^0.0.0",
    "@remirror/preset-core": "^0.0.0",
    "@remirror/react": "^0.13.1",
    "@remirror/react-utils": "^0.7.6",
    "@testing-library/react": "^10.0.4",
    "@types/node": "^14.0.5",
    "@types/react": "^16.9.35",
    "@types/react-dom": "^16.9.8",
    "@types/sanitize-html": "^1.23.0",
    "@types/testing-library__react": "^10.0.1",
    "jest-prosemirror": "^0.8.3",
    "jest-snapshot": "^26.0.1",
    "mutationobserver-shim": "^0.3.5",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "sanitize-html": "^1.24.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "preconstruct": {
    "entrypoints": [
      ".",
      "environment"
    ]
  }
}
