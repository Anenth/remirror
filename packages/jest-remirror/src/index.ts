export { ignoreJSDOMWarnings, jsdomExtras, jsdomPolyfill } from './jsdom-polyfills';
export { remirrorMatchers } from './jest-remirror-matchers';
export { renderEditor, RemirrorTestChain } from './jest-remirror-editor';
export { setupRemirrorEnvironment } from './jest-remirror-environment';
export type { RenderEditorParameter, TaggedProsemirrorNode } from './jest-remirror-types';

declare global {
  interface Window {
    hasWarnedAboutCancelAnimationFramePolyfill?: boolean;
    ignoreAllJSDOMWarnings: boolean;
  }
}
