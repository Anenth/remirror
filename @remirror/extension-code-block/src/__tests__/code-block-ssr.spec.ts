/**
 * @jest-environment node
 */

import { renderSSREditor } from 'jest-remirror';
import { CodeBlockExtension, CodeBlockExtensionOptions } from '../';

import javascript from 'refractor/lang/javascript';
import markdown from 'refractor/lang/markdown';
import typescript from 'refractor/lang/typescript';

const supportedLanguages = [typescript, javascript, markdown];
const create = (params: CodeBlockExtensionOptions = {}) =>
  renderSSREditor([new CodeBlockExtension({ ...params, supportedLanguages })], { initialContent });

test('ssr component', () => {
  expect(create()).toInclude('<pre class="language-markup"><code>Simple Code Blocks');
  expect(create()).toMatchSnapshot();
});

const initialContent = {
  type: 'doc',
  content: [
    {
      type: 'codeBlock',
      content: [
        {
          type: 'text',
          text:
            'Simple Code Blocks\necho "fun times"\nUse Shift-Enter or Mod-Enter to hard break out of the code block',
        },
      ],
    },
  ],
};
