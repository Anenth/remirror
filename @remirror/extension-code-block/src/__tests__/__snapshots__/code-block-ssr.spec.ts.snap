// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ssr component 1`] = `
"<style data-emotion-css=\\"1qat138\\">.css-1qat138 .remirror-editor{caret-color:currentColor;word-wrap:break-word;white-space:pre-wrap;overflow-wrap:break-word;}.css-1qat138 .remirror-editor:focus{outline:none;}.css-1qat138 code[class*='language-'],.css-1qat138 pre[class*='language-']{color:#c5c8c6;text-shadow:0 1px rgba(0,0,0,0.3);font-family:Inconsolata,Monaco,Consolas,'Courier New',Courier,monospace;direction:ltr;text-align:left;white-space:pre;word-spacing:normal;word-break:normal;line-height:1.5;-moz-tab-size:4;-o-tab-size:4;tab-size:4;-webkit-hyphens:none;-moz-hyphens:none;-ms-hyphens:none;-webkit-hyphens:none;-moz-hyphens:none;-ms-hyphens:none;hyphens:none;}.css-1qat138 pre[class*='language-']{padding:1em;margin:0.5em 0;overflow:auto;border-radius:0.3em;}.css-1qat138:not(pre) > code[class*='language-'],.css-1qat138 pre[class*='language-']{background:#1d1f21;}.css-1qat138:not(pre) > code[class*='language-']{padding:0.1em;border-radius:0.3em;}.css-1qat138 .token.comment,.css-1qat138 .token.prolog,.css-1qat138 .token.doctype,.css-1qat138 .token.cdata{color:#7c7c7c;}.css-1qat138 .token.punctuation{color:#c5c8c6;}.css-1qat138 .namespace{opacity:0.7;}.css-1qat138 .token.property,.css-1qat138 .token.keyword,.css-1qat138 .token.tag{color:#96cbfe;}.css-1qat138 .token.class-name{color:#ffffb6;-webkit-text-decoration:underline;text-decoration:underline;}.css-1qat138 .token.boolean,.css-1qat138 .token.constant{color:#99cc99;}.css-1qat138 .token.symbol,.css-1qat138 .token.deleted{color:#f92672;}.css-1qat138 .token.number{color:#ff73fd;}.css-1qat138 .token.selector,.css-1qat138 .token.attr-name,.css-1qat138 .token.string,.css-1qat138 .token.char,.css-1qat138 .token.builtin,.css-1qat138 .token.inserted{color:#a8ff60;}.css-1qat138 .token.variable{color:#c6c5fe;}.css-1qat138 .token.operator{color:#ededed;}.css-1qat138 .token.entity{color:#ffffb6;}.css-1qat138 .token.url{color:#96cbfe;}.css-1qat138 .language-css .token.string,.css-1qat138 .style .token.string{color:#87c38a;}.css-1qat138 .token.atrule,.css-1qat138 .token.attr-value{color:#f9ee98;}.css-1qat138 .token.function{color:#dad085;}.css-1qat138 .token.regex{color:#e9c062;}.css-1qat138 .token.important{color:#fd971f;}.css-1qat138 .token.important,.css-1qat138 .token.bold{font-weight:bold;}.css-1qat138 .token.italic{font-style:italic;}.css-1qat138 .token.entity{cursor:help;}</style><div class=\\"css-1qat138\\"><div role=\\"textbox\\" aria-multiline=\\"true\\" aria-label=\\"\\" class=\\"Prosemirror remirror-editor undefined\\" contenteditable=\\"true\\"><pre class=\\"language-markup\\"><code>Simple Code Blocks
echo &quot;fun times&quot;
Use Shift-Enter or Mod-Enter to hard break out of the code block</code></pre></div><div role=\\"textbox\\" aria-multiline=\\"true\\" aria-label=\\"\\" class=\\"Prosemirror remirror-editor undefined\\" contenteditable=\\"true\\"><pre class=\\"language-markup\\"><code>Simple Code Blocks
echo &quot;fun times&quot;
Use Shift-Enter or Mod-Enter to hard break out of the code block</code></pre></div></div>"
`;
